File: src/stack/Stack.java
package stack;

import stack.exceptions.EmptyStackException;
import stack.exceptions.FullStackException;

public class Stack {

    private int[] values;
    private int top;
    public Stack(){
        values = new int[10];
        top = -1;
    }

    public void push(int element){
        if(!isFull()){
            top++;
            values[top] = element;
        }else{
            throw new EmptyStackException("the stack is empty, stack something first");
        }
    }

    public boolean isEmpty(){
        return (top == -1);
    }

    public boolean isFull(){
        return (top == 9);
    }

    public int pop(){
        if(!isEmpty()){
            int elem = values[top];
            top--;
            return elem;
        }else {
            throw new FullStackException("the stack is full, pop something first");
        }

    }

}

File: src/main/stack/c/stack.c
#include "stack.h"

void create(struct Stack *s){
    s->top = -1;
}

void push(struct Stack *s, int element){
    s->top++;
    s->values[s->top] = element;
}

int pop(struct Stack *s){
    int element = s->values[s->top];
    s->top--;
    return element;
}

int isEmpty(struct Stack s){
    return (s.top == 1);
}

int isFull(struct Stack s){
    return (s.top == 9);
}



